google/apistore.go

```diff
@@ -85,3 +87,51 @@ func (c *APIStore) SetBucketAgeLifecycle(name string, days int64) error {
 	_, err := c.service.Buckets.Patch(name, bucket).Do()
 	return err
 }
+
+// GrantObjectViewer updates the IAM policy on the bucket to grant member the roles/storage.objectViewer role
+// The existing policy attributes on the bucket are preserved
+func (c *APIStore) GrantObjectViewer(bucket, member string) error {
+	return c.grantRole(bucket, member, "roles/storage.objectViewer")
+}
+
+// GrantObjectCreator updates the IAM policy on the bucket to grant member the roles/storage.objectCreator role
+// The existing policy attributes on the bucket are preserved
+func (c *APIStore) GrantObjectCreator(bucket, member string) error {
+	return c.grantRole(bucket, member, "roles/storage.objectCreator")
+}
+
+// GrantObjectAdmin updates the IAM policy on the bucket to grant member the roles/storage.objectAdmin role
+// The existing policy attributes on the bucket are preserved
+func (c *APIStore) GrantObjectAdmin(bucket, member string) error {
+	return c.grantRole(bucket, member, "roles/storage.objectAdmin")
+}
+
+func (c *APIStore) grantRole(bucket, member, role string) error {
+	existingPolicy, err := c.service.Buckets.GetIamPolicy(bucket).Do()
+	if err != nil {
+		return err
+	}
+
+	var added bool
+	for _, b := range existingPolicy.Bindings {
+		if b.Role == role {
+			for _, m := range b.Members {
+				if m == member {
+					// already granted
+					return nil
+				}
+			}
+			b.Members = append(b.Members, member)
+			added = true
```

Comment:
you should break here, unless there can be multiple `Bindings` with the same role.